File: coupon-common/src/main/java/com/kakao/mobility/kmcoupon/exception/CommonControllerAdvice.java
package com.kakao.mobility.kmcoupon.exception;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class CommonControllerAdvice {
    @ExceptionHandler(value = BadRequestException.class)
    public ResponseEntity<String> handleNotFoundException(BadRequestException e) {
        return ResponseEntity.badRequest().body(e.getMessage());
    }
}

File: coupon-common/src/main/java/com/kakao/mobility/kmcoupon/exception/CommonControllerAdvice.kt
package com.kakao.mobility.kmcoupon.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class CommonControllerAdvice {
    @ExceptionHandler(value = [BadRequestException::class])
    @ResponseStatus(HttpStatus.BAD_REQUEST)
    fun handleNotFoundException(e: BadRequestException): String? {
        return e.message
    }
}